fail-fast.md

```diff
@@ -0,0 +1,155 @@
+# Fail-Fast: Stop running tests on first failure
+
+There are two strategies for running CI tests. A developer either wants to get
+a comprehensive report on all failures in a test suite, or to get an fast
+feedback from the system and fail everything immediately on first failure.
+
+Semaphore by default runs all the jobs and prepares a comprehensive report for
+your pipeline. By setting a Fail-Fast strategy in your pipelines you can control
+this policy and get a fast response from the system.
+
+Two Fail-Fast strategies can be defined:
+
+- A 'stop' strategy that stops everything in your pipeline as soon as a failure
+  is detected.
+
+- A 'cancel' strategy that stops only the jobs and blocks in your pipeline that
+  haven't yet started. This option is ideal if you don't want to stop an already
+  started test execution.
+
+[Stopping all jobs in a pipeline on first failure]()
+[Stopping all jobs in a pipeline on first failure only on the master branch]()
+[Stop vs. Cancel strategies]()
+[See also](#see-also)
+
+## Stopping all jobs in a pipeline on first failure
+
+To stop all the jobs in a pipeline on first failure, define the fail-fast stop
+policy in your pipeline YAML files.
+
+``` yaml
+version: v1.0
+name: Tests
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+fail_fast:
+  stop:
+    when: "branch =~ '.*'"
```

Comment:
Maybe it is better to use `when: true` or `when: 'true'` here, otherwise it will not work for tags and pull requests.